src/pygenesis/find.py

```diff
@@ -0,0 +1,269 @@
+                        "Name:",
+                        object_dict["Content"],
+                        f"{'-' * 20}",
+                        "Information:",
+                        str(object_dict["Information"]),
+                        f"{'-' * 40}",
+                    ]
+                )
+
+            print(output)
+
+    @staticmethod
+    def _get_metadata_results(category: str, code: str):
+        """
+        Based on the category and code query parameters the metadata will be generated.
+
+        Args:
+            category (str): Category of the result. E.g. "tables", "cubes"
+            code (str): The code (identifier) of the relevant category object.
+        Returns:
+            response (json): The response as a json.
+        """
+        params = {
+            "name": code,
+        }
+
+        response = get_response_from_endpoint("metadata", category, params)
+        response_json = response.json()
+
+        return response_json
+
+
+class Find:
+    """
+    A class representing the find object that includes Result objects for variables, statistics, cubes and tables.
+
+    Attributes:
+        query (str): The query that is provided to find endpoint.
+        top_n_preview (int): Number of previews in print summary.
+
+    Methods:
+        _print_summary(): Prints a summary with result previews.
+        _get_find_results(): Gets results from the find endpoints.
+    """
+
+    def __init__(self, query: str, top_n_preview: int = 5) -> None:
+        """Method for retrieving data from find endpoint.
+
+        Args:
+            query (str): The query that is provided to find endpoint.
+            top_n_preview (int): Number of previews in print summary.
+
+        Returns:
+            summary: prints number of search results and firsts results
+            self.statistics: Statistics that match with the query.
+            self.tables: Tables that match with the query.
+            self.variables: Variables that match with the query.
+        """
+
+        self.query = query
+        self.top_n_preview = top_n_preview
+        self.statistics = self._get_find_results(query, "statistics")
+        self.variables = self._get_find_results(query, "variables")
+        self.tables = self._get_find_results(query, "tables")
+        self.cubes = self._get_find_results(query, "cubes")
+
+        print(self._print_summary())
+
+    def _print_summary(self):
+        return "\n".join(
+            [
+                "##### Results #####",
+                f"{'-' * 40}",
+                f"# Number of tables: {len(self.tables.df)}",
+                "# Preview:",
+                self.tables.df.iloc[0 : self.top_n_preview].to_markdown(),
+                f"{'-' * 40}"
+                f"# Number of statistics: {len(self.statistics.df)}",
+                "# Preview:",
+                self.statistics.df.iloc[0 : self.top_n_preview].to_markdown(),
+                f"{'-' * 40}",
```

Comment:
hier braucht es keinen f-String, '-' * 40 geht auch so :)

Reply:
Pylint besteht leider auf f-Strings.

Reply:
Ach echt? Owohl in 220 auch ein normaler String verwendet wird? Das ist ja spannend, in dem Falle ist pylint sicherlich schlauer!